openstates/de/committees.py

```diff
@@ -8,104 +8,87 @@ class DECommitteeScraper(CommitteeScraper,LXMLMixin):
             self.scrape_joint_committees(term,session)
 
         url = urls[chamber] % (session,)
-        page = lxml.html.fromstring(self.get(url).text)
-        page.make_links_absolute(url)
-
-        for row in page.xpath('//tr'):
-            if len(row.xpath('./td')) > 0:
-                #if statement removes header tr
-                comm = row.xpath('.//a')[1]
-                comm_name = comm.text_content().strip()
-
-                comm_url = comm.attrib["href"]
-
-                comm_page = lxml.html.fromstring(self.get(comm_url).text)
-                comm_page.make_links_absolute(comm_url)
-                committee = Committee(chamber, comm_name)
-                committee.add_source(comm_url)
-                committee.add_source(url)
-
-                chair = comm_page.xpath(".//div[@class='sub_title']")
-                chair = chair[0].text.replace("Chairman:","").strip()
-                committee.add_member(chair,"Chairman")
-
-                for table in comm_page.xpath(".//table"):
-                    header,content = table.xpath(".//td")
-                    header = header.text_content().strip()
-                    content = content.text_content().strip()
-                    if "Vice" in header:
-                        if content:
-                            committee.add_member(content,"Vice-Chairman")
-                    elif header == "Members:":
-                        for m in content.split("\n"):
-                            committee.add_member(m.strip())
-
-                self.save_committee(committee)
+        data = self.post(url).json()['Data']
+
+        for item in data:
+            committee = Committee(chamber, item['CommitteeName'])
+            chair_man = str(item.get('ChairName', 'No'))
+            vice_chair = str(item.get('ViceChairName', 'No'))
+            comm_id = item['CommitteeId']
+            comm_url = 'http://legis.delaware.gov/CommitteeDetail?' + \
+                        'committeeId='+ str(comm_id)
+            comm_page = lxml.html.fromstring(self.get(comm_url).text)
+            #all members including chair_man and vice_chair
+            members = comm_page.xpath("//section[@class='section-short']/" + \
+                                      "div[@class='info-horizontal']/div" + \
+                                      "[@class='info-group']/div[@class" + \
+                                      "='info-value']/a/text()")
+
+            committee.add_member(chair_man,'Chairman')
+            committee.add_member(vice_chair,'Vice-Chair')
+            for member in members:
+                #vice_chair and chair_man already added.
+                if chair_man not in member and vice_chair not in member:
+                    committee.add_member(member)
+
+            committee.add_source(comm_url)
+            committee.add_source(url)
+
+            self.save_committee(committee)
+
 
     def scrape_joint_committees(self,term,session):
```

Comment:
Actually, in URLs variable, I added 'Joint' 
```
urls = {
            'upper': 'http://legis.delaware.gov/json/Committees/' +
                     'GetCommitteesByTypeId?assemblyId=%s&committeeTypeId=1',
            'lower': 'http://legis.delaware.gov/json/Committees/' +
                     'GetCommitteesByTypeId?assemblyId=%s&committeeTypeId=2',
           'joint': 'http://legis.delaware.gov/json/Committees/' + 
              'GetCommitteesByTypeId?assemblyId=%s&committeeTypeId=3'
        }
```

`url = urls[chamber] % (session,)`

 But on running scraper, only `upper` and `lower` were values of `chamber` variable. I tried it by adding `joint` in  __init__.py file's `chamber` dictionary but nothing happened. that's why I used `scrape_joint_committees` method. This is the same technique which was used before to scrape the `joint` committees.